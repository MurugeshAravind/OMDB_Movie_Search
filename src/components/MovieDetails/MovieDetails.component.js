import React from "react";
import { useEffect, useState } from "react";
import { Link, useParams } from "react-router-dom";
import styled from "styled-components";
import FetchData from "../../common/Api/FetchData";
import Loader from "../../common/Loader/Loader";
import * as AppConstants from "../../AppConstants";
const Heading = styled.h1`
  text-align: center;
`;
const Post = styled.img.attrs((props) => ({
  src: props.src || AppConstants.DUMMY_IMAGE_PATH,
}))`
  alt: "Poster";
  background-color: black;
  border: solid 1px black;
  @media screen and (max-width: 600px) {
    display: block;
    margin: 0 auto;
    padding: 0;
    width: 175px;
  }
`;
const Button = styled.button`
  background-color: #ccc;
  border: none;
  color: black;
  padding: 10px 20px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  margin: 0.5rem;
  border-radius: 16px;
  clear: right;
  @media screen and (max-width: 600px) {
    display: block;
    margin: 1rem auto;
  }
  @media screen and (min-width: 600px) {
    display: block;
    margin: 1rem auto;
  }
`;
const WrapperDiv = styled.div`
  display: block;
  margin: 2rem;
  text-align: center;
  @media screen and (max-width: 600px) {
    display: block;
    text-align: center;
  }
  @media screen and (min-width: 600px) {
    display: block;
    text-align: center;
  }
`;
const MovieDetails = () => {
  const { id } = useParams("id");
  const [movieDetails, setMovieDetails] = useState(null);
  const [isLoading, setIsLoading] = useState(false);
  useEffect(() => {
    async function fetchMovieDetails() {
      setIsLoading(true);
      let movieDetails = await FetchData(
        `${AppConstants.URL}${AppConstants.API_KEY}&i=${id}`
      );
      setIsLoading(false);
      if (
        movieDetails.data.Response === "True" &&
        Object.keys(movieDetails.data).length > 0
      ) {
        setMovieDetails(movieDetails.data);
      }
    }
    fetchMovieDetails();
  }, [id]);
  return !isLoading && movieDetails && Object.keys(movieDetails).length > 0 ? (
    <>
      <div data-testid="details">
        <WrapperDiv>
          <Link to={{ pathname: "/OMDB_Movie_Search" }}>
            <big>
              <strong>Home</strong>
            </big>
          </Link>
          <Heading>{movieDetails.Title}</Heading>
        </WrapperDiv>
        <WrapperDiv>
          <Post
            src={
              movieDetails.Poster !== "N/A"
                ? movieDetails.Poster
                : AppConstants.DUMMY_IMAGE_PATH
            }
          />
        </WrapperDiv>

        <WrapperDiv>
          <p>
            <strong>Year:</strong>&nbsp;
            {<Button key={movieDetails.Year}>{movieDetails.Year}</Button>}
          </p>
          <p>
            <strong>Rated:</strong>&nbsp;
            {<Button key={movieDetails.Rated}>{movieDetails.Rated}</Button>}
          </p>
          <p>
            <strong>Genre:</strong>&nbsp;
            {movieDetails.Genre.split(",").map((x) => (
              <Button key={x}>{x}</Button>
            ))}
          </p>
          <p>
            <strong>Plot:</strong>&nbsp;{movieDetails.Plot}
          </p>
          <p>
            <strong>Actors:</strong>&nbsp;
            {movieDetails.Actors.split(",").map((x) => (
              <Button key={x}>{x}</Button>
            ))}
          </p>
          <p>
            <strong>Director:</strong>&nbsp;
            <Button>{movieDetails.Director}</Button>
          </p>
          <p>
            <strong>Writer/ Writers:</strong>&nbsp;
            {movieDetails.Writer.split(",").length ? (
              movieDetails.Writer.split(",").map((x) => (
                <Button key={x}>{x}</Button>
              ))
            ) : (
              <Button>{movieDetails.Writer}</Button>
            )}
          </p>
          <p>
            <strong>
              Awards:&nbsp;<Button>{movieDetails.Awards}</Button>
            </strong>
          </p>
          <p>
            <label htmlFor="rating">
              <strong>OMDB Rating:</strong>
            </label>
            <meter
              id="rating"
              name="name"
              min=""
              low="3"
              high="9"
              max="10"
              value={movieDetails.imdbRating}
            ></meter>
            &nbsp; {movieDetails.imdbRating}
          </p>
        </WrapperDiv>
      </div>
    </>
  ) : (
    <Loader />
  );
};
export default MovieDetails;
