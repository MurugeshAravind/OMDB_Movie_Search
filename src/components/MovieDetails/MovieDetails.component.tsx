import { useEffect, useState } from 'react';
import { Link, useParams } from 'react-router-dom';
import FetchData from '../../common/Api/FetchData';
import Loader from '../../common/Loader/Loader';
import * as AppConstants from '../../AppConstants';

interface MovieDetailsInterface {
  Title?: string;
  Poster?: string;
  Year?: string;
  Rated?: string;
  Genre?: string;
  Actors?: string;
  Director?: string;
  Writer?: string;
  imdbRating?: string;
  Plot?: string;
  Awards?: string;
}

const MovieDetails = () => {
  const { id } = useParams();
  const [movieDetails, setMovieDetails] = useState<MovieDetailsInterface | any>(
    [],
  );
  const [isLoading, setIsLoading] = useState<boolean>(false);

  useEffect(() => {
    async function fetchMovieDetails() {
      setIsLoading(true);
      let movieDetails = await FetchData({
        method: 'GET',
        url: `${AppConstants.URL}${AppConstants.API_KEY}&i=${id}`,
      });
      setIsLoading(false);
      if (
        movieDetails?.data.Response === 'True' &&
        Object.keys(movieDetails?.data).length > 0
      ) {
        setMovieDetails(movieDetails?.data);
      }
    }
    fetchMovieDetails();
  }, [id]);

  return !isLoading && movieDetails && Object.keys(movieDetails).length > 0 ? (
    <div data-testid="details">
      <div className="block m-8 text-center">
        <Link
          to="/OMDB_Movie_Search/"
          className="text-blue-600 hover:underline text-lg font-bold"
        >
          Home
        </Link>
        <h1 className="text-center text-[2rem] text-silver font-bold mt-4 mb-8">
          {movieDetails?.Title}
        </h1>
      </div>
      <div className="block m-8 text-center">
        <img
          src={
            movieDetails?.Poster !== 'N/A'
              ? movieDetails?.Poster
              : AppConstants.DUMMY_IMAGE_PATH
          }
          alt="Poster"
          className="mx-auto bg-black border border-black rounded-lg w-[300px] max-w-full sm:w-[175px]"
        />
      </div>
      <div className="block m-8 text-center">
        <p>
          <strong>Year:</strong>&nbsp;
          <span className="inline-block bg-gray-300 text-black px-4 py-2 rounded-2xl m-2">
            {movieDetails?.Year}
          </span>
        </p>
        <p>
          <strong>Rated:</strong>&nbsp;
          <span className="inline-block bg-gray-300 text-black px-4 py-2 rounded-2xl m-2">
            {movieDetails?.Rated}
          </span>
        </p>
        <p>
          <strong>Genre:</strong>&nbsp;
          {movieDetails?.Genre?.split(',').length > 0 ? (
            movieDetails?.Genre?.split(',').map((x: any) => (
              <span
                key={x}
                className="inline-block bg-gray-300 text-black px-4 py-2 rounded-2xl m-2"
              >
                {x}
              </span>
            ))
          ) : (
            <span className="inline-block bg-gray-300 text-black px-4 py-2 rounded-2xl m-2">
              {movieDetails?.Genre}
            </span>
          )}
        </p>
        <p>
          <strong>Plot:</strong>&nbsp;{movieDetails?.Plot}
        </p>
        <p>
          <strong>Actors:</strong>&nbsp;
          {movieDetails?.Actors?.split(',').length > 0 ? (
            movieDetails?.Actors?.split(',').map((x: any) => (
              <span
                key={x}
                className="inline-block bg-gray-300 text-black px-4 py-2 rounded-2xl m-2"
              >
                {x}
              </span>
            ))
          ) : (
            <span className="inline-block bg-gray-300 text-black px-4 py-2 rounded-2xl m-2">
              {movieDetails?.Actors}
            </span>
          )}
        </p>
        <p>
          <strong>Director/ Directors:</strong>&nbsp;
          {movieDetails?.Director?.split(',').length > 0 ? (
            movieDetails?.Director?.split(',').map((x: any) => (
              <span
                key={x}
                className="inline-block bg-gray-300 text-black px-4 py-2 rounded-2xl m-2"
              >
                {x}
              </span>
            ))
          ) : (
            <span className="inline-block bg-gray-300 text-black px-4 py-2 rounded-2xl m-2">
              {movieDetails?.Director}
            </span>
          )}
        </p>
        <p>
          <strong>Writer/ Writers:</strong>&nbsp;
          {movieDetails?.Writer?.split(',').length > 0 ? (
            movieDetails?.Writer?.split(',').map((x: any) => (
              <span
                key={x}
                className="inline-block bg-gray-300 text-black px-4 py-2 rounded-2xl m-2"
              >
                {x}
              </span>
            ))
          ) : (
            <span className="inline-block bg-gray-300 text-black px-4 py-2 rounded-2xl m-2">
              {movieDetails?.Writer}
            </span>
          )}
        </p>
        <p>
          <strong>
            Awards:&nbsp;
            <span className="inline-block bg-gray-300 text-black px-4 py-2 rounded-2xl m-2">
              {movieDetails?.Awards}
            </span>
          </strong>
        </p>
        <p>
          <strong>OMDB Rating:</strong>
          <span className="inline-block bg-gray-300 text-black px-4 py-2 rounded-2xl m-2">
            <meter
              id="rating"
              min=""
              low={3}
              high={9}
              max={10}
              value={movieDetails?.imdbRating}
            ></meter>
            &nbsp;{`${movieDetails?.imdbRating}`}
          </span>
        </p>
      </div>
    </div>
  ) : (
    <Loader />
  );
};

export default MovieDetails;
