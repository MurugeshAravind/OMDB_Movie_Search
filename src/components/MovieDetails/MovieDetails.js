import { useEffect, useState } from "react";
import { Link, useParams } from "react-router-dom";
import styled from "styled-components";
import FetchData from "../Api/FetchData";
import Loader from "../Loader/Loader";
import * as AppConstants from "../../AppConstants";
const Heading = styled.h1`
  text-align: center;
  @media screen and (max-width: 600px) {
    font-size: 20px;
  }
`;
const Post = styled.img.attrs((props) => ({
  src: props.src || "",
}))`
  height: auto;
  alt: "Poster";
  float: left;
  border: solid 1px black;
  margin-right: 1rem;
  padding: 0.5rem;
  background-color: black;
  @media screen and (max-width: 600px) {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
  }
`;
const Button = styled.button`
  background-color: #ccc;
  border: none;
  color: black;
  padding: 10px 20px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  margin: 0.5rem;
  border-radius: 16px;
  clear: right;
  @media screen and (max-width: 600px) {
    display: block;
    margin: 1rem auto;
  }
`;
const WrapperDiv = styled.div`
  @media screen and (max-width: 600px) {
    display: block;
    text-align: center;
    margin: 40rem auto;
  }
`;
const MovieDetails = () => {
  const { id } = useParams("id");
  console.log(id)
  const [movieDetails, setMovieDetails] = useState(null);
  const [isLoading, setIsLoading] = useState(false);
  useEffect(() => {
    async function fetchMovieDetails() {
      setIsLoading(true);
      let movieDetails = await FetchData(
        `${AppConstants.URL}${AppConstants.API_KEY}&i=${id}`
      )
        .then((res) => res)
        .then((res) => res.json());

      setIsLoading(false);
      if (
        movieDetails.Response === "True" &&
        Object.keys(movieDetails).length > 0
      ) {
        console.log("movieDetails", movieDetails);
        setMovieDetails(movieDetails);
      }
    }
    fetchMovieDetails();
  }, [id]);
  return !isLoading && movieDetails && Object.keys(movieDetails).length > 0 ? (
    <>
      <Link to="/" style={{ fontStyle: "unset", float: "left" }}>
        <big>
          <strong>Back</strong>
        </big>
      </Link>
      <Heading>
        {movieDetails.Title}
        <pre>
          <strong>{movieDetails.Year}</strong>* &nbsp;
          <strong>{movieDetails.Rated}</strong>* &nbsp;
          <strong>{movieDetails.Runtime}</strong>* &nbsp;
        </pre>
      </Heading>
      <Post src={movieDetails.Poster} />
      <WrapperDiv>
        <p>
          <strong>Genre:</strong>&nbsp;
          {movieDetails.Genre.split(",").map((x) => (
            <Button key={x}>{x}</Button>
          ))}
        </p>
        <p>
          <strong>Plot:</strong>&nbsp;{movieDetails.Plot}
        </p>
        <p>
          <strong>Actors:</strong>&nbsp;
          {movieDetails.Actors.split(",").map((x) => (
            <Button key={x}>{x}</Button>
          ))}
        </p>
        <p>
          <strong>Director:</strong>&nbsp;
          <Button>{movieDetails.Director}</Button>
        </p>
        <p>
          <strong>Writer/ Writers:</strong>&nbsp;
          {movieDetails.Writer.split(",").length ? (
            movieDetails.Writer.split(",").map((x) => (
              <Button key={x}>{x}</Button>
            ))
          ) : (
            <Button>{movieDetails.Writer}</Button>
          )}
        </p>
        <p>
          <strong>
            Rating:&nbsp;<Button>{movieDetails.imdbRating}</Button>
          </strong>
        </p>
        <p>
          <strong>
            Awards:&nbsp;<Button>{movieDetails.Awards}</Button>
          </strong>
        </p>
      </WrapperDiv>
    </>
  ) : (
    <Loader />
  );
};
export default MovieDetails;
