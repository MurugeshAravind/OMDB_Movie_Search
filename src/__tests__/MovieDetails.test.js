import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import MovieDetails from "../components/MovieDetails/MovieDetails.component";
const MovieDetailsComponent = renderer.create(<MovieDetails />).toJSON();
describe("movie details test cases", () => {
  test("movie detail snapshot test", () => {
    expect(MovieDetailsComponent).toMatchSnapshot();
  })
});
