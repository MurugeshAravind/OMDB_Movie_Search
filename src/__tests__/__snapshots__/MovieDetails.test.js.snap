// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movie details test cases movie detail snapshot test 1`] = `
<div
  className="dhVqPt"
/>
`;
